File: microservices/common-lib/src/main/java/com/trafficfine/common/dto/PaymentResponse.java
package com.trafficfine.common.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record PaymentResponse(
        String paymentReference,
        String referenceNumber,
        BigDecimal amount,
        String status,
        LocalDateTime paidAt,
        String message
) {}

File: android-app/app/src/main/java/com/example/trafficfinesystem/data/PaymentResponse.kt
package com.example.trafficfinesystem.data

import com.google.gson.annotations.SerializedName

data class PaymentResponse(
    @SerializedName("paymentReference") val paymentReference: String,
    @SerializedName("referenceNumber") val referenceNumber: String,
    @SerializedName("amount") val amount: Double,
    @SerializedName("status") val status: String,
    @SerializedName("paidAt") val paidAt: String,
    @SerializedName("message") val message: String
)
